### ISSUE TITLE
Improve build speed by splitting tests

### ISSUE THREAD

**Issue by sdelamo** [micronaut-projects/micronaut-starter#207]
> one test in particular is holding up the whole thing which is CreateAwsLambdaSpec. @Sergio del Amo Caballero can you split this into different specs so they have a chance to run in parallel. Right now all the other tests compete but that one takes like 4 minutes extra on its own

**Comment by graemerocher** [comment]
I suggest as well we move all of the integration tests to a particular package and then have 2 separate jobs on github actions, one that runs the integration tests and one that runs the unit tests and does checkstyle/spotless etc.

We can use the `--tests` flag in Gradle which allows patterns to include tests only under `io.micronaut.starter.integration.*` or only under ``io.micronaut.starter.unit.*`

**Comment by sdelamo** [comment]
I've pushed two options function tests in individual gradle modules:  

https://github.com/micronaut-projects/micronaut-starter/commit/75913a1df9e69fd4ce5a0b56c364b7a4e96c5729

or function tests in two single modules `test-core` and `test-clip`. Those are the functional tests of `starter-core` and `starter-cli`. 

The fastest approach would be for each individual module instal of having a class such as: 

```
package io.micronaut.starter.core.test

import io.micronaut.starter.application.ApplicationType
import io.micronaut.starter.test.CommandSpec
import io.micronaut.starter.test.LanguageBuildCombinations
import io.micronaut.starter.options.BuildTool
import io.micronaut.starter.options.Language
import spock.lang.Unroll

class CreateCliSpec extends CommandSpec {

    @Unroll
    void 'create-cli-app with #lang and #buildTool'(Language lang, BuildTool buildTool) {
        given:
        ApplicationType applicationType = ApplicationType.CLI
        generateProject(lang, buildTool, [], applicationType)

        when:
        if (buildTool == BuildTool.GRADLE) {
            executeGradleCommand('run --args="-v"')
        } else {
            executeMavenCommand("mn:run -Dmn.appArgs=-v")
        }

        then:
        testOutputContains("Hi")

        where:
        [lang, buildTool] << LanguageBuildCombinations.combinations()
    }
}
``` 

have multiple classes: 

A class for Java/Gradle

```java
package io.micronaut.starter.core.test

import io.micronaut.starter.application.ApplicationType
import io.micronaut.starter.test.CommandSpec
import io.micronaut.starter.test.LanguageBuildCombinations
import io.micronaut.starter.options.BuildTool
import io.micronaut.starter.options.Language
import spock.lang.Unroll

class CreateCliJavaGradleSpec extends CommandSpec {

    @Unroll
    void 'create-cli-app with #lang and #buildTool'(Language lang, BuildTool buildTool) {
        given:
        ApplicationType applicationType = ApplicationType.CLI
        generateProject(lang, buildTool, [], applicationType)

        when:
        if (buildTool == BuildTool.GRADLE) {
            executeGradleCommand('run --args="-v"')
        } else {
            executeMavenCommand("mn:run -Dmn.appArgs=-v")
        }

        then:
        testOutputContains("Hi")

        where:
        [lang, buildTool] << [Language.JAVA, BuildTool.GRADLE]
    }
}

A class for Java/Maven

```java
package io.micronaut.starter.core.test

import io.micronaut.starter.application.ApplicationType
import io.micronaut.starter.test.CommandSpec
import io.micronaut.starter.test.LanguageBuildCombinations
import io.micronaut.starter.options.BuildTool
import io.micronaut.starter.options.Language
import spock.lang.Unroll

class CreateCliJavaMavenSpec extends CommandSpec {

    @Unroll
    void 'create-cli-app with #lang and #buildTool'(Language lang, BuildTool buildTool) {
        given:
        ApplicationType applicationType = ApplicationType.CLI
        generateProject(lang, buildTool, [], applicationType)

        when:
        if (buildTool == BuildTool.GRADLE) {
            executeGradleCommand('run --args="-v"')
        } else {
            executeMavenCommand("mn:run -Dmn.appArgs=-v")
        }

        then:
        testOutputContains("Hi")

        where:
        [lang, buildTool] << [Language.JAVA, BuildTool.MAVEN]
    }
}
``` 

A class for Kotlin/Maven
